File: kvt-projekat2024/src/app/models/Manages.ts
import { User } from "./User";
import { Facility } from "./Facility";

export class Manages {
  id: number;
  startDate: Date;
  endDate: Date;
  user: User;
  facility: Facility;

  constructor(id: number, startDate: Date, endDate: Date, user: User, facility: Facility) {
    this.id = id;
    this.startDate = startDate;
    this.endDate = endDate;
    this.user = user;
    this.facility = facility;
  }
}

File: src/main/java/rs/ac/uns/ftn/svt/model/Manages.java
package rs.ac.uns.ftn.svt.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Data
@Entity
public class Manages {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate startDate;

    private LocalDate endDate;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "facility_id", nullable = false)
    private Facility facility;

}
